core/include/gnuradio-4.0/Scheduler.hpp

```diff
@@ -17,24 +19,29 @@ using gr::thread_pool::BasicThreadPool;
 
 enum ExecutionPolicy { singleThreaded, multiThreaded };
 
-template<profiling::ProfilerLike TProfiler = profiling::null::Profiler>
+constexpr std::chrono::milliseconds kMessagePollInterval{ 10 };
+
+template<typename Derived, profiling::ProfilerLike TProfiler = profiling::null::Profiler>
 class SchedulerBase {
-public:
```

Comment:
Since you made already larger modification to this class, please replace the custom state handling with `SchedulerBase` publically deriving from the  `lifecycle::StateMachine`. This should simplify also the callback handle w.r.t. `start()`, `stop()` and others and make this more consistent with `Block<T>`.

Reply:
This causes a heavy amount of untested code-duplication (see below) and must be addressed/simplified before being merged.

Reply:
Done, refactored the schedulers accordingly.